### DISCUSSION TITLE
fortls unexpected path behaviour for vscode

### DISCUSSION THREAD

**Discussion by breengles** [fortran-lang/fortls#267] [Q&A] [answered]
Hi!

Fortls is able to start if 
```
"fortran.fortls.path": "/Users/breengles/mambaforge/bin/fortls"
```
but it fails with any of the following:
```
"fortran.fortls.path": "~/mambaforge/bin/fortls"
```
```
"fortran.fortls.path": "{env:HOME}/mambaforge/bin/fortls"
```
```
"fortran.fortls.path": "{userHome}/mambaforge/bin/fortls"
```
that is only if the path is fully and explicitly specified `fortls` starts on macOS.

Are there any ideas how to specify it via env variables as I work not only on mac but also on linux and it would be really nice to have this setting to be synced :)

versions:
fortls 2.13.0
modern fortran 3.2.0
vscode 1.76.1 (Universal)

**Comment by gnikit** [comment, marked as answer]
So currently the vscode extension does not have access to the ${env} environment.
We should really fix that. If you don't want to install fortls on your base Python distribution/ don't want the pip packages to be in the path
I would maybe suggest using a Python virtual environment that is local to your project. Then the fortls path would always be
`${workspaceFolder}/venv/bin/fortls`

**Reply by gnikit** [reply]
As a side note, try the Modern Fortran pre-release. There are a series of fixes and features there.

**Comment by breengles** [comment]
Hi again mates!

Here I come again :)

First of all, let me thank you all for the answers!

I hope that someday it will become possible to specify either with 
```"fortran.fortls.path": "~/mambaforge/bin/fortls"``` 
or 
```"fortran.fortls.path": "${env:HOME}/mambaforge/bin/fortls"```.
For instance, with `black` both 
```"python.formatting.blackPath": "~/mambaforge/bin/black"```
and
```"python.formatting.blackPath": "${env:HOME}/mambaforge/bin/black"```
work just fine.

Though I wrote all the things above, the following is working on the pre-release version of the Modern Fortran extension
```"fortran.fortls.path": "${env:HOME}/mambaforge/bin/fortls"```. 
Hopefully, it will be fully released soon, thanks!
